### PR TITLE
[14.0] shopfloor_manual_product_transfer: new warning about reserved qty

### PR DESCRIPTION
**Pull request by sebalix**
... and display processed quantity with the `packaging-qty-picker` in readonly mode + some minor improvements/fixes.

image

Ref: rau-67

### CODE REVIEW

File: shopfloor_manual_product_transfer/services/manual_product_transfer.py [hunk 1]
```diff
@@ -511,9 +521,17 @@ def _unreserve_other_lines(self, location, move_lines):
...
+                # add 'product_id' in the domain here? We could block the entire
+                # process here (see check on 'extra_move_lines' below) because
+                # another move line for another product is found in the given
+                # location... is it what we want?
             ]
         )
         extra_move_lines = location_move_lines - move_lines
+        # TODO: question, why do we care about lines that could be related to
+        # other products here? Should it prevent us to unreserve the moves of
+        # the product we're interested in?
```
Comment (sebalix) [inline shopfloor_manual_product_transfer/services/manual_product_transfer.py]: ping @jbaudoux I would like your insight on this, I think `location_move_lines` could contain unrelated lines (of other products sharing the same location), maybe it doesn't happen IRL with location bins but to me it's a bug, unless I didn't understood what was the purpose of the unreserve process in this scenario.

Comment (jbaudoux) [reply shopfloor_manual_product_transfer/services/manual_product_transfer.py]: Indeed the method should take in argument an option product. For the location content transfer, it is correct to unreserve all others moves as we empty the location. For this scenario where we move only a product, we should limit to the given product.

Comment (sebalix) [reply shopfloor_manual_product_transfer/services/manual_product_transfer.py]: Fixed here:
- https://github.com/OCA/wms/pull/482

File: shopfloor_manual_product_transfer/services/manual_product_transfer.py [hunk 2]
```diff
@@ -304,6 +324,17 @@ def _check_quantity(self, location, product, lot, quantity):
                 message=self.msg_store.qty_exceeds_initial_qty(),
             )
 
+    def _check_quantity_in_stock(self, location, product, quantity, lot=None):
+        """Check if there is enough quantity of a product in the location."""
+        current_qty = self._get_product_qty(location, product, lot, free=True)
+        initial_qty = self._get_initial_qty(location, product, lot)
+        return (
+            # FIXME use float_compare
```
Comment (TDu) [inline shopfloor_manual_product_transfer/services/manual_product_transfer.py]: Probably a good time to fix it.

File: shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js [hunk 3]
```diff
@@ -66,9 +76,11 @@ const ManualProductTransfer = {
 
             <div class="detail item-detail-card" v-if="state_is('scan_destination_location')">
                 <v-card :color="utils.colors.color_for('screen_step_done')">
-                    <v-card-title>
-                        <span class="label">Quantity = {{ quantity() }}</span>
-                    </v-card-title>
+                    <packaging-qty-picker
+                        :key="make_state_component_key(['packaging-qty-picker', 1232])"
```
Comment (simahawk) [inline shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js]: 1232?? this should be a uniq value. Use the id of the record if you have one

File: shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js [hunk 4]
```diff
@@ -130,6 +142,16 @@ const ManualProductTransfer = {
...
         },
+        warning: function () {
+            const data = this.state.data;
+            if (_.isEmpty(data)) {
+                return "";
+            }
+            if (!_.isEmpty(data.warning)) {
+                return data.warning;
+            }
+            return "";
```
Comment (simahawk) [inline shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js]:
```suggestion
            return _.result(this.state, "data.warning", "")
```

File: shopfloor_manual_product_transfer/actions/message.py [hunk 5]
```diff
@@ -25,3 +25,12 @@ def transfer_canceled_success(self, picking):
             "message_type": "success",
             "body": _("Transfer {} canceled.").format(picking.name),
         }
+
+    def qty_assigned_to_preserve(self, product, qty_assigned):
+        return {
+            "message_type": "warning",
+            "body": _(
+                f"{qty_assigned} {product.uom_id.name} are reserved in this "
```
Comment (simahawk) [inline shopfloor_manual_product_transfer/actions/message.py]: Are you sure `f` strings work fine w/ translations?

File: shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js (hunk 4 above)
Comment (simahawk) [reply shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js]: BTW IMO this warning should be defined only frontend side and enabled w/ a flag instead.

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by simahawk** [comment]
And please shorten the commits that are too long ;)

File: shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js (hunk 4 above)
Comment (sebalix) [reply shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js]: I think about it, and also think about the translations which are already integrated in the backend (not there are no translations in Shopfloor but it's easier on the backend).

File: shopfloor_manual_product_transfer/actions/message.py (hunk 5 above)
Comment (sebalix) [reply shopfloor_manual_product_transfer/actions/message.py]: Not at all :D fixed

File: shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js (hunk 3 above)
Comment (sebalix) [reply shopfloor_manual_product_transfer_mobile/static/wms/src/scenario/manual_product_transfer.js]: Fixed, everywhere + tested locally.

File: shopfloor_manual_product_transfer/services/manual_product_transfer.py (hunk 2 above)
Comment (sebalix) [reply shopfloor_manual_product_transfer/services/manual_product_transfer.py]: Fixed

**Review by simahawk — APPROVED** [review]

**Review by TDu — APPROVED** [review]
